src/org/zaproxy/zap/extension/jsonView/ExtensionHttpPanelJsonView.java

```diff
@@ -0,0 +1,213 @@
+		panelManager.removeResponseViews(ResponseSplitComponent.NAME, HttpPanelJsonView.NAME, ResponseSplitComponent.ViewComponent.BODY);
+		//remove default view selectors and their factories
+		panelManager.removeRequestDefaultViewSelectorFactory(
+				RequestSplitComponent.NAME,
+				JsonDefaultViewSelectorFactory.NAME);
+		//TODO change to removeRequestDefaultViewSelectors on next major release
+		panelManager.removeRequestDefaultViewSelectorFactoryAndDefaultViewSelectorsAdded(
+				ResponseSplitComponent.NAME,
+				JsonDefaultViewSelector.NAME,
+				ResponseSplitComponent.ViewComponent.BODY);
+		panelManager.removeResponseDefaultViewSelectorFactory(
+				ResponseSplitComponent.NAME,
+				JsonDefaultViewSelectorFactory.NAME);
+		panelManager.removeResponseDefaultViewSelectors(
+				ResponseSplitComponent.NAME,
+				JsonDefaultViewSelector.NAME,
+				ResponseSplitComponent.ViewComponent.BODY);
+	}
+
+	private static final class RequestJsonViewFactory implements HttpPanelViewFactory {
+		public static final String NAME = "RequestJsonViewFactory";
+
+		@Override
+		public String getName() { return NAME; }
+
+		@Override
+		public HttpPanelView getNewView() {
+			return new HttpPanelJsonView(new RequestBodyStringHttpPanelViewModel());
+		}
+
+		@Override
+		public Object getOptions() { return RequestSplitComponent.ViewComponent.BODY; }
+	}
+
+	private static final class ResponseJsonViewFactory implements HttpPanelViewFactory {
+		public static final String NAME = "ResponseJsonViewFactory";
+
+		@Override
+		public String getName() {
+			return NAME;
+		}
+
+		@Override
+		public HttpPanelView getNewView() {
+			return new HttpPanelJsonView(new ResponseBodyStringHttpPanelViewModel());
+		}
+
+		@Override
+		public Object getOptions() {
+			return ResponseSplitComponent.ViewComponent.BODY;
+		}
+	}
+
+	private static final class JsonDefaultViewSelector implements HttpPanelDefaultViewSelector {
+		private static final String NAME = "JsonDefaultViewSelector";
+		public Object options;
+
+		public JsonDefaultViewSelector(Object options){
+			this.options = options;
+		}
+
+		@Override
+		public String getName() {
+			return NAME;
+		}
+
+		@Override
+		public boolean matchToDefaultView(Message message) {
+			if(message == null){
+				//called with null when deleting the view, ie. when changing from split to combined view
+				return false;
+			}
+			if(message instanceof HttpMessage){
+				HttpMessage httpMessage = (HttpMessage) message;
+				HttpHeader header;
+				if(this.options == RequestSplitComponent.NAME) {
+					header = httpMessage.getRequestHeader();
+				} else if (this.options == ResponseSplitComponent.NAME){
+					header = httpMessage.getResponseHeader();
+				} else {
+					return false;
+				}
+				if(header.getHeader(HttpHeader.CONTENT_TYPE) == null){ return false; }
+				if(header.getHeader(HttpHeader.CONTENT_TYPE).toLowerCase().contains("application/json")){ return true; }

```

Comment:
make this the default view only for application/json messages that parse as valid json (provided by HttpPanelJsonView.isEnabled)